pkg/operator/controller/ingress/metrics.go

```diff
@@ -26,9 +26,15 @@ var (
 		Help: "Report the conditions for ingress controllers. 0 is False and 1 is True.",
 	}, []string{"name", "condition"})
 
+	activeNLBs = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "ingress_controller_aws_nlb_active",
+		Help: "Report the number of active NLBs on AWS clusters. 0 is False, 1 is True",
+	}, []string{"name"})
```

Comment:
As mentioned in https://github.com/openshift/cluster-monitoring-operator/pull/1509 the `name` label is potentially problematic, as it seems to have an unbounded value set.
Consider dropping this to keep cardinality of this metric in check.

Reply:
If you remove the name, please validate what the metric looks like when one ingress-controller has NLB, but another one does not.  We should still see a metric with the value 1.

Reply:
within a cluster, the amount of ingress controllers is not that high (2 at max?).
In openshift/cluster-monitoring-operator#1509  I changed it to a sum across the whole cluster, so you only get a single integer back without a label. I hope that's OK @jan--f?

Reply:
Sure the recording rule is fine, however we do ingest this metric into the in-cluster stack.
That being said, a dozen or so label values is perfectly fine. I suppose creating those is nothing the average user can do right?
The only concern here would be that someone can create lot of label values in a short time frame (by creating a lot of ingress controllers), blowing up the resources needed to track this metric. Often upgrades can be an issue too where things get restarted and names change rapidly because they are (at least partly) autogenerated. None of that seems to be a concern here though so :+1:

Reply:
yeah, I think the user will be in bigger trouble by creating the equivalent of ingress controllers before Prometheus would blow up on the metric dimension. :+1: